### PR TITLE
Automatically validate DEPENDENCIES.md 

### PR DESCRIPTION
**Pull request by ordovicia**
This PR adds `scripts/check_dep_list.py`, which validates that `DEPENDENCIES.md` is synced with `Gopkg.toml`.
The validation is performed automatically in CI.

### CODE REVIEW

File: scripts/check_dep_list.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        print("Dependencies in " + GOPKG_PATH + ":")
+        for g in gopkg:
+            print("-", g)
+        print()
+
+    deplist = deps_list()
+    if verbose:
+        print("Dependencies in " + DEPLIST_PATH + ":")
+        for d in deplist:
+            print("-", d)
```
Comment (dtaniwaki) [inline scripts/check_dep_list.py]: can we print them with something like `print("- %40s %20s" % d)` for better readability?

File: scripts/check_dep_list.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+WHITELIST_DEPLIST = [  # may be missing in DEPENDENCIES.md
+]
+
+
+def main(verbose=False):
+    ok = True
+
+    gopkg = deps_gopkg()
+    if verbose:
+        print("Dependencies in " + GOPKG_PATH + ":")
```
Comment (dtaniwaki) [inline scripts/check_dep_list.py]: I think it would be easier to read by sorting values for both lists. Or, maybe printing diff is helpful.

**Comment by ordovicia** [comment]
Fixed the output of `check_dep_list.py`.
if `Gopkg.toml` and `DEPENDENCIES.md` differ, the script prints the difference in side-by-side style.

```
$ ./scripts/check_dep_list.py -v
Dependencies written in Gopkg.toml
- github.com/cpuguy83/strongerrors
- github.com/pkg/errors
- github.com/sirupsen/logrus
- github.com/spf13/cobra
- github.com/spf13/viper
- gopkg.in/yaml.v2
- k8s.io/client-go
- k8s.io/kubernetes

Dependencies written in DEPENDENCIES.md
- github.com/containerd/containerd         Apache-2.0
- github.com/cpuguy83/strongerrors         Apache-2.0
- github.com/pkg/errors                    BSD-2-Clause
- github.com/sirupsen/logrus               MIT
- github.com/stretchr/testify              MIT
- gopkg.in/yaml.v2                         Apache-2.0
- k8s.io/api                               Apache-2.0
- k8s.io/apimachinery                      Apache-2.0
- k8s.io/client-go                         Apache-2.0
- k8s.io/kubernetes                        Apach'-2.0

Difference between Gopkg.toml and DEPENDENCIES.md
github.com/spf13/cobra                   <
github.com/spf13/viper                   <
                                         > github.com/containerd/containerd
                                         > github.com/stretchr/testify
```

**Comment by ordovicia** [comment]
merged master.

**Pull request merged**